alerting/src/test/kotlin/org/opensearch/alerting/resthandler/MonitorRestApiIT.kt

```diff
@@ -1102,4 +1085,77 @@ class MonitorRestApiIT : AlertingRestTestCase() {
+
+        assertEquals("Create monitor failed", RestStatus.CREATED, createResponse.restStatus())
+        val responseBody = createResponse.asMap()
+        val createdId = responseBody["_id"] as String
+        val createdVersion = responseBody["_version"] as Int
+        assertNotEquals("response is missing Id", Monitor.NO_ID, createdId)
+        assertTrue("incorrect version", createdVersion > 0)
+        val actualLocation = createResponse.getHeader("Location")
+        assertEquals("Incorrect Location header", "$ALERTING_BASE_URI/$createdId", actualLocation)
+    }
+
+    @Throws(Exception::class)
+    fun `test getting a document level monitor`() {
+        val monitor = createRandomDocumentMonitor()
+
+        val storedMonitor = getMonitor(monitor.id)
+
+        assertEquals("Indexed and retrieved monitor differ", monitor, storedMonitor)
+    }
+
+    @Throws(Exception::class)
+    fun `test updating conditions for a document level monitor`() {
+        val monitor = createRandomDocumentMonitor()
+
+        val updatedTriggers = listOf(
+            QueryLevelTrigger(
+                name = "foo",
+                severity = "1",
+                condition = Script("return true"),
+                actions = emptyList()
+            )
+        )
+        val updateResponse = OpenSearchRestTestCase.client().makeRequest(
+            "PUT", monitor.relativeUrl(),
+            emptyMap(), monitor.copy(triggers = updatedTriggers).toHttpEntity()
+        )
+
+        OpenSearchRestTestCase.assertEquals("Update monitor failed", RestStatus.OK, updateResponse.restStatus())
+        val responseBody = updateResponse.asMap()
+        OpenSearchRestTestCase.assertEquals("Updated monitor id doesn't match", monitor.id, responseBody["_id"] as String)
+        OpenSearchRestTestCase.assertEquals("Version not incremented", (monitor.version + 1).toInt(), responseBody["_version"] as Int)
+
+        val updatedMonitor = getMonitor(monitor.id)
+        OpenSearchRestTestCase.assertEquals("Monitor trigger not updated", updatedTriggers, updatedMonitor.triggers)
+    }
+
+    @Throws(Exception::class)
+    fun `test deleting a document level monitor`() {
+        val monitor = createRandomDocumentMonitor()
+
+        val deleteResponse = client().makeRequest("DELETE", monitor.relativeUrl())
+        assertEquals("Delete failed", RestStatus.OK, deleteResponse.restStatus())
+
+        val getResponse = client().makeRequest("HEAD", monitor.relativeUrl())
+        assertEquals("Deleted monitor still exists", RestStatus.NOT_FOUND, getResponse.restStatus())
+    }
+
+    @Throws(Exception::class)
+    fun `test creating a document monitor with error trigger`() {
+        val trigger = randomQueryLevelTrigger()
+        val monitor = randomDocumentLevelMonitor(triggers = listOf(trigger))
+        try {
+            client().makeRequest("POST", ALERTING_BASE_URI, emptyMap(), monitor.toHttpEntity())
+        }catch (e:java.lang.Exception){
```

Comment:
After the call, add a `fail` assertion stating an exception should have been thrown since document level monitors do not support non-document level triggers.
Example of [fail usage](https://github.com/opensearch-project/alerting/blob/main/alerting/src/test/kotlin/org/opensearch/alerting/MonitorTests.kt#L20)